interpretation/deepseismic_interpretation/azureml_pipelines/README.md

```diff
@@ -0,0 +1,174 @@
+Define parameters for the run in a config file. See an example [here](../pipeline_config.json)
+```json
+{
+    "step1":
+    {
+        "type": "<type of step. Supported types include PythonScriptStep and MpiStep>",
+        "name": "<name in AzureML for this step>",
+        "script": "<path to script for this step>",
+        "output": "<name of the output in AzureML for this step - optional>",
+        "input_datareference_path": "<path on the data reference for the input data - optional>",
+        "input_datareference_name": "<name of the data reference in AzureML where the input data lives - optional>",
+        "input_dataset_name": "<name of the datastore in AzureML - optional>",
+        "source_directory": "<source directory containing the files for this step>",
+        "arguments": "<arguments to pass to the script - optional>",
+        "requirements": "<path to the requirements.txt file for the step - optional>",
+        "node_count": "<number of nodes to run the script on - optional>",
+        "processes_per_node": "<number of processes to run on each node - optional>",
+        "base_image": "<name of an image registered on dockerhub that you want to use as your base image"
+    },
+  
+    "step2":
+    {
+        .
+        .
+        .
+    }
+}
+```
+  
+## Kicking off a Pipeline
+In order to kick off a pipeline, you will need to use the AzureCLI to login to the subscription where your workspace resides:
+```bash
+az login
+az account set -s <subscription id>
+```
+Kick off the training pipeline defined in your config via your python environment of choice. The code will look like this:
+```python
+from src.azml.train_pipeline.train_pipeline import TrainPipeline
+  
+orchestrator = TrainPipeline("<path to your config file>")
+orchestrator.construct_pipeline()
+run = orchestrator.run_pipeline(experiment_name="DEV-train-pipeline")
+```
+See an example in [dev/kickoff_train_pipeline.py](dev/kickoff_train_pipeline.py)
+
+If this fails due to access to the Azure ML subscription, you may be able to connect by using a workaround:
+Go to [base_pipeline.py](../base_pipeline.py) and add the following import:
+```python
+from azureml.core.authentication import AzureCliAuthentication
+```
+Then find the code where we connect to the workspace which looks like this:
+```python
+self.ws = Workspace.from_config(path=ws_config)
+```
+and replace it with  this:
+```python
+cli_auth = AzureCliAuthentication()
+self.ws = Workspace(subscription_id=<subscription id>, resource_group=<resource group>, workspace_name=<workspace name>, auth=cli_auth)
+```
+to get this to run, you will also need to `pip install azure-cli-core`
+Then you can go back and follow the instructions above, including az login and setting the subscription, and kick off the pipeline.
+  
+## Cancelling a Pipeline Run
+If you kicked off a pipeline and want to cancel it, run the [cancel_run.py](dev/cancel_run.py) script with the corresponding run_id and step_id.
```

Comment:
Where do we get these IDs from and how?

Reply:
Updated README.md to include this information in this commit https://github.com/microsoft/seismic-deeplearning/pull/195/commits/40496bebf941100917a167b89afc0d5d08342364. Once you run the script it will print these IDs but you will also be able to find the IDs and cancel the run directly in the portal. Resolving this conversation.